whois-commons/src/main/java/net/ripe/db/whois/common/JvmSecurityManager.java

```diff
@@ -0,0 +1,19 @@
+package net.ripe.db.whois.common;
+
+import net.ripe.db.whois.common.profiles.DeployedProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@DeployedProfile
+public class JvmSecurityManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JvmSecurityManager.class);
+
+    @Autowired
+    public JvmSecurityManager() {
+        java.security.Security.setProperty("networkaddress.cache.ttl", "30");
```

Comment:
Can we set this property at runtime? Will it make a difference?

The OpenJDK code suggests this property is only read once (it's in a static block):
https://github.com/AdoptOpenJDK/openjdk-jdk/blob/jdk-11+28/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L120

Reply:
Stack overflow to the rescue:

`This code must execute before any other code in the JVM attempts to perform networking operations.`

Is it enough to also put these properties in a static block?

https://github.com/AdoptOpenJDK/openjdk-jdk/blob/jdk-11+28/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L120

Reply:
A workaround to not setting the security properties is to set the *system* properties `sun.net.inetaddr.ttl` and `sun.net.inetaddr.negative.ttl`. They are deprecated but work for now.

https://stackoverflow.com/questions/41708885/best-way-to-set-networkaddress-cache-ttl

Reply:
There is another way to provide security file but let's have a discussion about it, we need to think about implications of it.
https://dzone.com/articles/how-override-java-security

Reply:
Lets try not to use the deprecated methods first if we can find better way to do this